chia/rpc/wallet_rpc_api.py

```diff
@@ -3457,161 +3427,116 @@ async def nft_get_info(self, request: dict[str, Any]) -> EndpointResult:
-        wallet_id = uint32(request["wallet_id"])
-        nft_wallet = self.service.wallet_state_manager.get_wallet(id=wallet_id, required_type=NFTWallet)
-        royalty_address = request.get("royalty_address", None)
-        if isinstance(royalty_address, str) and royalty_address != "":
-            royalty_puzhash = decode_puzzle_hash(royalty_address)
-        elif royalty_address in {None, ""}:
+        nft_wallet = self.service.wallet_state_manager.get_wallet(id=request.wallet_id, required_type=NFTWallet)
+        if request.royalty_address in {None, ""}:
             royalty_puzhash = await action_scope.get_puzzle_hash(self.service.wallet_state_manager)
         else:
-            royalty_puzhash = bytes32.from_hexstr(royalty_address)
-        royalty_percentage = request.get("royalty_percentage", None)
-        if royalty_percentage is None:
-            royalty_percentage = uint16(0)
-        else:
-            royalty_percentage = uint16(int(royalty_percentage))
+            assert request.royalty_address is not None  # hello mypy
+            royalty_puzhash = decode_puzzle_hash(request.royalty_address)
         metadata_list = []
-        for meta in request["metadata_list"]:
-            if "uris" not in meta.keys():
-                return {"success": False, "error": "Data URIs is required"}
-            if not isinstance(meta["uris"], list):
-                return {"success": False, "error": "Data URIs must be a list"}
-            if not isinstance(meta.get("meta_uris", []), list):
-                return {"success": False, "error": "Metadata URIs must be a list"}
-            if not isinstance(meta.get("license_uris", []), list):
-                return {"success": False, "error": "License URIs must be a list"}
+        for meta in request.metadata_list:
             nft_metadata = [
-                ("u", meta["uris"]),
-                ("h", hexstr_to_bytes(meta["hash"])),
-                ("mu", meta.get("meta_uris", [])),
-                ("lu", meta.get("license_uris", [])),
-                ("sn", uint64(meta.get("edition_number", 1))),
-                ("st", uint64(meta.get("edition_total", 1))),
+                ("u", meta.uris),
+                ("h", meta.hash),
+                ("mu", meta.meta_uris),
+                ("lu", meta.license_uris),
+                ("sn", meta.edition_number),
+                ("st", meta.edition_total),
             ]
-            if "meta_hash" in meta and len(meta["meta_hash"]) > 0:
-                nft_metadata.append(("mh", hexstr_to_bytes(meta["meta_hash"])))
-            if "license_hash" in meta and len(meta["license_hash"]) > 0:
-                nft_metadata.append(("lh", hexstr_to_bytes(meta["license_hash"])))
+            if meta.meta_hash is not None:
+                nft_metadata.append(("mh", meta.meta_hash))
+            if meta.license_hash is not None:
+                nft_metadata.append(("lh", meta.license_hash))
```

Comment:
a couple more cases where the interface is no longer accepting an empty string.  is that a case we need to support?